Unreal/CarlaUE4/Plugins/CarlaTools/Source/CarlaTools/Private/OpenDriveToMap.cpp

```diff
@@ -154,16 +154,59 @@ void UOpenDriveToMap::CreateMap()
     UE_LOG(LogCarlaToolsMapGenerator, Error, TEXT("Map Name Is Empty") );
     return;
   }
-  if ( !IsValid(FileDownloader) )
-  {
-    FileDownloader = NewObject<UCustomFileDownloader>();
-  }
 
-  FileDownloader->ResultFileName = MapName;
-  FileDownloader->Url = Url;
+  if( !Url.IsEmpty() ) {
+    if ( !IsValid(FileDownloader) )
+    {
+      FileDownloader = NewObject<UCustomFileDownloader>();
+    }
 
-  FileDownloader->DownloadDelegate.BindUObject( this, &UOpenDriveToMap::ConvertOSMInOpenDrive );
-  FileDownloader->StartDownload();
+    FileDownloader->ResultFileName = MapName;
+    FileDownloader->Url = Url;
+
+    FileDownloader->DownloadDelegate.BindUObject( this, &UOpenDriveToMap::ConvertOSMInOpenDrive );
+    FileDownloader->StartDownload();
+  }else{
```

Comment:
### Clean Code
This else should unified with the next line making an `else if`.
Whe should reduce the number of nested scopes when possible to improve redability.

Reply:
Done